### PR TITLE
Added ability to change entryFee to SocialTradingManager.

### PR DESCRIPTION
**Pull request by bweick**

### CODE REVIEW

File: contracts/managers/SocialTradingManager.sol [hunk 1]
```diff
@@ -342,6 +416,24 @@ contract SocialTradingManager is
...
+    function validateNewEntryFee(
+        uint256 _entryFee
+    )
+        internal
+        view
+    {
+        require(
+            _entryFee <= maxEntryFee,
+            "SocialTradingManager.validateNewEntryFee: Passed entry fee must not exceed maxEntryFee."
+        );
```
Comment (felix2feng) [inline contracts/managers/SocialTradingManager.sol]: Do we want to validate that it's a multiple of a basis point? If so, you can import `ScaleValidations` in set-protocol-contracts

File: contracts/managers/SocialTradingManager.sol [hunk 2]
```diff
@@ -200,9 +220,63 @@ contract SocialTradingManager is
...
     }
 
+    /*
+     * Start fee update process, set fee update timestamp and commit to new fee.
+     *
+     * @param _tradingPool        The address of the trading pool being updated
+     * @param _newEntryFee        New entry fee in a scaled decimal value
+     *                              (e.g. 100% = 1e18, 1% = 1e16)
+     */
+    function initiateSetEntryFee(
```
Comment (felix2feng) [inline contracts/managers/SocialTradingManager.sol]: maybe a clearer name is `initiateEntryFeeChange` and `finalizeEntryFeeChange`

File: contracts/managers/SocialTradingManager.sol [hunk 3]
```diff
@@ -200,9 +220,63 @@ contract SocialTradingManager is
...
+            block.timestamp >= feeUpdateTimestamp(_tradingPool),
+            "SocialTradingManager.finalizeSetFeeRecipient: Time lock period must elapse to update fees."
+        );
+
+        // Save entryFee to memory and reset fee related attributes
+        uint256 newEntryFee = newEntryFee(_tradingPool);
+        pools[address(_tradingPool)].feeUpdateTimestamp = 0;
+        pools[address(_tradingPool)].newEntryFee = 0;
+
+        _tradingPool.setEntryFee(newEntryFee);
```
Comment (felix2feng) [inline contracts/managers/SocialTradingManager.sol]: Why not just do the following in one line: `_tradingPool.setEntryFee(newEntryFee(_tradingPool)`?

File: contracts/managers/SocialTradingManager.sol (hunk 1 above)
Comment (felix2feng) [reply contracts/managers/SocialTradingManager.sol]: Nm, that would be redundant to the setEntryFee on the RBSet

File: contracts/managers/SocialTradingManager.sol (hunk 3 above)
Comment (bweick) [reply contracts/managers/SocialTradingManager.sol]: I guess that one should be fine, updating the timestamp should be enough to stop any reentrancy type attack

**Pull request merged**